addons/sms/tests/test_sms_template_access_rights.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests.common import SavepointCase, new_test_user
+from odoo.exceptions import AccessError
+
+
+class TestSmsTemplateAccessRights(SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.sms_manager = new_test_user(cls.env, login='sms_manager', groups='base.group_system')
+        cls.sms_user = new_test_user(cls.env, login='sms_user', groups='base.group_user')
+
+    def create_sms_template(self, user, model):
+        return self.env['sms.template'].with_user(user).create(dict({
+            'name': 'SMS Template ' + model.name,
+            'body': 'Body Test',
+            'model_id': model.id,
+        }))
+
+    def read_sms_template(self, user, templates, field):
+        data = templates.with_user(user).read([field])
+        if len(templates) == 1:
+            return data[0][field]
+        mapped_data = {record['id']: record for record in data}
+        return [mapped_data[template_id][field] for template_id in templates.ids]
+
+    def test_access_rights_sms_template(self):
+        models = self.env['ir.model'].search(['&', ('is_mail_thread_sms', '=', True), ('transient', '=', False)])
+        user_admin = self.env.ref('base.user_admin')
+        for model in models:
+            sms_template = self.create_sms_template(self.sms_manager, model)
+            # Test CRUD for group_user
+            self.assertEqual(self.read_sms_template(self.sms_user, sms_template, 'name'), 'SMS Template ' + model.name, "A member of group_user should be able to read a template")
```

Comment:
You just have to read a field and check it returns a value.
No need to check equality with actual value, you could just `assertTrue(bool(result))`.

Also, you always pass "name", so let's not make it a method argument.
And your read method should be private.
I don't think you will even need a read method when you refactor to one test for each user:
`self.assertTrue(bool(sms_template.name))` should be enough.

Reply:
done